src/CodeGenerator/FactoryGenerator.php

```diff
@@ -7,24 +7,38 @@
 
 namespace Zend\Di\CodeGenerator;
 
-use Psr\Container\ContainerInterface;
-use Zend\Code\Generator\ClassGenerator;
-use Zend\Code\Generator\DocBlockGenerator;
-use Zend\Code\Generator\FileGenerator;
-use Zend\Code\Generator\MethodGenerator;
-use Zend\Code\Generator\ParameterGenerator;
+use SplFileObject;
 use Zend\Di\ConfigInterface;
-use Zend\Di\Resolver\AbstractInjection;
+use Zend\Di\Exception\RuntimeException;
 use Zend\Di\Resolver\DependencyResolverInterface;
+use Zend\Di\Resolver\InjectionInterface;
 use Zend\Di\Resolver\TypeInjection;
 
+use function file_get_contents;
+use function strrpos;
+use function strtr;
+use function substr;
+
 /**
  * Generates factory classes
  */
 class FactoryGenerator
 {
     use GeneratorTrait;
 
+    private const TEMPLATE_FILE = __DIR__ . '/../../templates/factory.template';
+    private const PARAMETERS_TEMPLATE = <<<__CODE__
```

Comment:
If you use [nowdoc syntax](http://www.php.net/manual/en/language.types.string.php#language.types.string.syntax.nowdoc), you can eliminate the escaping in this template.

Additionally, use a space before the identifier:

```php
private const PARAMETERS_TEMPLATE = <<< '__CODE__'
        if (empty($options)) {
            $args = [
                %s
            ];
        } // etc.
__CODE__;
```